oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/OptimizerStateMachine.kt

```diff
@@ -0,0 +1,43 @@
+package com.empowerops.volition.ref_oasis
+
+import com.empowerops.volition.ref_oasis.State.*
+
+enum class State {
+    Idle,
+    StartPending,
+    Running,
+    PausePending,
+    Paused,
+    StopPending
+}
+
+class OptimizerStateMachine {
+    var currentState: State = Idle
+    private val stateTable: Map<State, List<State>> = mapOf(
+            Idle to listOf(StartPending),
+            StartPending to listOf(Running, Idle),
+            Running to listOf(PausePending, StopPending),
+            PausePending to listOf(Paused, StopPending),
+            Paused to listOf(Running, StopPending),
+            StopPending to listOf(Idle)
+    )
```

Comment:
During pending stop(stopping) state, since we are still waiting the result response. The reason we are waiting is we don't manage plugin state, so we are expecting a result which will make sure the plugin is at the state of being able to run again. This is very sequential to make sure when we request evaluation, and when we evaluate them again that plugin is not at the state of `Not ready to evaluate`
We can, add this check or call back to update its state by doing:
- keep track the state and wait for a call back. If we issue cancel, we just assume the plugin is cancel issued, when plugin calling back and tell us "I am done canceling", we mark it as ready but before that when we need to evaluation it again, we are going to check plugin readiness when we need to evaluate them and do the evaluation when ready. It will be implemented like awaitReady() we talked about.